<!DOCTYPE html>
<html lang="en">

<head>

    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="description" content="">
    <meta name="author" content="">
    <link rel="icon" href="images/rnb_favicon.ico">

    <title>RnBeads</title>

    <!-- Bootstrap Core CSS -->
    <link href="css/bootstrap.min.css" rel="stylesheet">

    <!-- Custom CSS -->
    <link href="css/small-business.css" rel="stylesheet">

</head>

<body>

    <!-- Navigation -->
    <nav class="navbar navbar-default navbar-fixed-top" role="navigation">
        <div class="container">
            <!-- Brand and toggle get grouped for better mobile display -->
            <div class="navbar-header">
                <button type="button" class="navbar-toggle" data-toggle="collapse" data-target="#bs-example-navbar-collapse-1">
                    <span class="sr-only">Toggle navigation</span>
                    <span class="icon-bar"></span>
                    <span class="icon-bar"></span>
                    <span class="icon-bar"></span>
                </button>
                <a class="navbar-brand" href="index.html">
                    <img src="images/rnb_logo_145x50.png" alt="">
                </a>
            </div>
            <!-- Collect the nav links, forms, and other content for toggling -->
            <div class="collapse navbar-collapse" id="bs-example-navbar-collapse-1">
                <ul class="nav navbar-nav">
                    <li class="active">
                        <a href="index.html">About</a>
                    </li>
                    <li>
                        <a href="installation.html">Installation</a>
                    </li>
                    <li>
                        <a href="tutorial.html">Tutorials</a>
                    </li>
                    <li>
                        <a href="examples.html">Examples</a>
                    </li>
                    <li class="dropdown">
                      <a class="dropdown-toggle" data-toggle="dropdown" href="#">Resources
                      <span class="caret"></span></a>
                      <ul class="dropdown-menu">
                        <li><a href="methylomes.html">Methylome Resource</a></li>
                        <li><a href="regions.html">Region Sets</a></li>
                        <li><a href="ageprediction.html">Age Prediction</a></li>
                      </ul>
                    </li>
                    <li>
                        <a href="references.html">References</a>
                    </li>
                    <li>
                        <a href="faq.html">FAQ</a>
                    </li>
<!--                     <li>
                        <a href="webservice.html">Webservice</a>
                    </li> -->
                    <li>
                        <a href="contact.html">Contact</a>
                    </li>
                </ul>
            </div>
            <!-- /.navbar-collapse -->
        </div>
        <!-- /.container -->
    </nav>

    <!-- Page Content -->
    <div class="container">

        <!-- Heading Row -->
        <div class="row">
            <div class="col-md-12">
                <img class="img-responsive img-rounded" src="images/rnb_workflow_1800x605.png" alt="">
            </div>
            <!-- /.col-md-8 -->
            <div class="col-md-12">
                <h1>Comprehensive Analysis of DNA Methylation Data with RnBeads</h1>
                <p>RnBeads is an R package for comprehensive analysis of DNA methylation data obtained with any experimental protocol that provides single-CpG resolution. Supported assays include Infinium and EPIC microarrays and bisulfite sequencing protocols, and also MeDIP-seq and MBD-seq once the data have been preprocessed with DNA methylation level inference software. RnBeads implements an analysis workflow that is significantly more comprehensive than those of existing tools. It documents its results in a highly annotated and readable hypertext report, and it scales to the large sample sizes that are becoming the norm for DNA methylation analysis in human cohorts.</p>
                <p>Various DNA methylation assays and input formats are supported. The pipeline implements state-of-the-art normalization techniques. Experimental quality control can be conducted and sample outliers and mix-ups can be identified. The package provides flexible methods for CpG and sample filtering. According to sample annotations, batch effects and phenotype covariates can be identified. DNA methylation distributions are analyzed and intergroup as well as intragroup variability in methylation profiles is quantified. Furthermore, differential methylation between groups of samples can be characterized. The analysis is based on individual CpGs as well as on predefined or custom genomic regions. Finally, methylation data can be exported in various formats including genome browser views. Comprehensive, highly interpretable reports containing method descriptions, publication grade plots and data tables are generated. Their HTML format facilitates easy tracking and comparison of analyses as well as exchanging results with collaboration partners. Due to its modularized concept, both first-time users and experts can conveniently perform analyses according to their individual demands. A single, comprehensive analysis run can be invoked by specifying only few parameters and executing a master command. Alternatively, a user may execute the steps of the pipeline individually.</p>
            </div>
            <!-- /.col-md-4 -->
        </div>
        <!-- /.row -->

        <hr>

        <!-- Call to Action Well -->
        <div class="row">
            <div class="col-md-12">
                <div class="well">
                <h2 style="color:red">RnBeads 2.0 published</h2>
                <p style="color:red">Happy Pi-day 2019. We are happy to announce that a manuscript describing RnBeads 2.0 has been published in <a href="https://doi.org/10.1186/s13059-019-1664-9">Genome Biology (2019).</a></p>
                </div>
            </div>
            <!-- /.col-lg-12 -->
        </div>
        <div class="row">
            <div class="col-md-12">
                <div class="well">
                <h2>New GUI</h2>
                <p>The new <a href="http://bioconductor.org/packages/release/bioc/html/RnBeads.html">RnBeads version</a> provides a graphical user interface that makes configuring and running RnBeads analysis more accessible to new and experienced users. The <a href="http://medical-epigenomics.org/software/rnbeads/materials/data/tutorial/RnBeadsDJ/RnBeads_GUI_quickstart.pdf">GUI vignette</a> and the <a href="https://rnbeads.org/tutorial.html">tutorial page</a> contain detailed descriptions on how to use this powerful tool. In brief, to start the GUI simply load the package and run the command:</p>
<pre class="pre-scrollable">
library(RnBeads)
rnb.run.dj()
</pre>
                </div>
            </div>
            <!-- /.col-lg-12 -->
        </div>
        <!-- /.row -->
        <div class="row">
            <div class="col-md-12">
                <div class="well">
                    RnBeads is available as an open source R package. We encourage user feedback and contributions. 
                    If you have suggestions for improving RnBeads or if you run into technical problems, please feel free to contact us at <a href="mailto:team@rnbeads.org">team@rnbeads.org</a> - ideally with a detailed description and test case illustrating your point.
                </div>
            </div>
            <!-- /.col-lg-12 -->
        </div>
        <!-- /.row -->

        <!-- Content Row -->
        <div class="row">
            <div class="col-md-4">
                <h2>Bioconductor</h2>
                <p>RnBeads is part of <a href="http://bioconductor.org/packages/release/bioc/html/RnBeads.html">Bioconductor</a>. You can install the package and obtain the source code from there.</p>
                <!-- <a class="btn btn-primary" href="http://bioconductor.org/packages/release/bioc/html/RnBeads.html">View RnBeads on Bioconductor</a> -->
            </div>
            <!-- /.col-md-4 -->
            <div class="col-md-4">
                <h2>Publication</h2>
                <p>A manuscript describing RnBeads 2.0 has been published in <a href="https://doi.org/10.1186/s13059-019-1664-9">Genome Biology (2019)</a>. The software was initially published in <a href="https://www.nature.com/articles/nmeth.3115">Nature Methods (2014).</a></p>
                <!-- <a class="btn btn-primary" href="http://www.nature.com/nmeth/journal/vaop/ncurrent/full/nmeth.3115.html">View Article</a> -->
            </div>
            <!-- /.col-md-4 -->
            <div class="col-md-4">
                <h2>Documentation</h2>
                <p>The full documentation is available in the <a href="http://bioconductor.org/packages/release/bioc/vignettes/RnBeads/inst/doc/RnBeads.pdf">package vignette</a> and the <a href="http://bioconductor.org/packages/release/bioc/manuals/RnBeads/man/RnBeads.pdf">reference manual</a>.</p>
            </div>
            <!-- /.col-md-4 -->
        </div>
        <!-- /.row -->
        <!-- Content Row -->
        <div class="row extra-top-padding">
            <div class="col-md-12">
                <h2>Supporting Institutes</h2>
                <p>RnBeads was developed by the <a href="http://computational-epigenetics.de/">Computational Epigenetics Groups</a> at the <a href="https://www.mpi-inf.mpg.de/home/">Max Planck Institute for Informatics</a> and <a href="http://epigenetik.uni-saarland.de">Saarland University</a>. Its developers now also hold positions at the <a href="http://cemm.at/research/groups/christoph-bock-group/">CeMM Research Center for Molecular Medicine of the Austrian Academy of Sciences</a> and the <a href="https://www.dkfz.de/en/CanEpi/">German Cancer Research Center</a>. RnBeads is also supported by the German Network for Bioinformatics Infrastructure (<a href="https://www.denbi.de/">de.NBI</a>).</p>
            </div>
        </div>
        <!-- /.row -->

        <hr>

        <div class="row">
            <div class="col-md-1"></div>
            <div class="col-md-2">
                <img class="img-responsive" src="images/mpii_150x75px.png" alt="">
            </div>
            <div class="col-md-2">
                <img class="img-responsive" src="images/uds_150x75px.png" alt="">
            </div>
            <div class="col-md-2">
                <img class="img-responsive" src="images/cemm_150x75px.png" alt="">
            </div>
            <div class="col-md-2">
                <img class="img-responsive" src="images/dkfz_150x75px.png" alt="">
            </div>
            <div class="col-md-2">
                <img class="img-responsive" src="images/denbi_150x75px.png" alt="">
            </div>
            <div class="col-md-1"></div>
        </div>
        <!-- /.row -->

    </div>
    <!-- /.container -->

    <!-- Footer -->
    <hr>
    <footer class="py-5 bg-inverse">
        <div class="container">>           <p class="m-0 text-center text-white">Copyright &copy; <a href="mailto:team@rnbeads.org">RnBeads Development Team</a> 2019 | <a href="https://cemm.at/contact/disclaimer/">Disclaimer / Impressum / Data protection</a></p>
        </div>
    </footer>

    <!-- jQuery -->
    <script src="js/jquery.js"></script>

    <!-- Bootstrap Core JavaScript -->
    <script src="js/bootstrap.min.js"></script>

</body>

</html>
