<!DOCTYPE html>
<html lang="en">

<head>

    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="description" content="">
    <meta name="author" content="">
    <link rel="icon" href="images/rnb_favicon.ico">

    <title>RnBeads</title>

    <!-- Bootstrap Core CSS -->
    <link href="css/bootstrap.min.css" rel="stylesheet">

    <!-- Custom CSS -->
    <link href="css/small-business.css" rel="stylesheet">

</head>

<body>

    <!-- Navigation -->
    <nav class="navbar navbar-default navbar-fixed-top" role="navigation">
        <div class="container">
            <!-- Brand and toggle get grouped for better mobile display -->
            <div class="navbar-header">
                <button type="button" class="navbar-toggle" data-toggle="collapse" data-target="#bs-example-navbar-collapse-1">
                    <span class="sr-only">Toggle navigation</span>
                    <span class="icon-bar"></span>
                    <span class="icon-bar"></span>
                    <span class="icon-bar"></span>
                </button>
                <a class="navbar-brand" href="index.html">
                    <img src="images/rnb_logo_145x50.png" alt="">
                </a>
            </div>
            <!-- Collect the nav links, forms, and other content for toggling -->
            <div class="collapse navbar-collapse" id="bs-example-navbar-collapse-1">
                <ul class="nav navbar-nav">
                    <li>
                        <a href="index.html">About</a>
                    </li>
                    <li>
                        <a href="installation.html">Installation</a>
                    </li>
                    <li>
                        <a href="tutorial.html">Tutorials</a>
                    </li>
                    <li>
                        <a href="examples.html">Examples</a>
                    </li>
                    <li class="dropdown">
                      <a class="dropdown-toggle" data-toggle="dropdown" href="#">Resources
                      <span class="caret"></span></a>
                      <ul class="dropdown-menu">
                        <li><a href="methylomes.html">Methylome Resource</a></li>
                        <li><a href="regions.html">Region Sets</a></li>
                        <li><a href="ageprediction.html">Age Prediction</a></li>
                      </ul>
                    </li>
                    <li class="active">
                        <a href="references.html">References</a>
                    </li>
                    <li>
                        <a href="faq.html">FAQ</a>
                    </li>
<!--                     <li>
                        <a href="webservice.html">Webservice</a>
                    </li> -->
                    <li>
                        <a href="contact.html">Contact</a>
                    </li>
                </ul>
            </div>
            <!-- /.navbar-collapse -->
        </div>
        <!-- /.container -->
    </nav>

    <!-- Page Content -->
    <div class="container">

        <!-- Page Heading/Breadcrumbs -->
        <h1 class="mt-4 mb-3">References <!-- <small>Subheading</small> --></h1>

        <!-- 
        <ol class="breadcrumb">
            <li class="breadcrumb-item">
                <a href="index.html">Home</a>
            </li>
            <li class="breadcrumb-item active">Contact</li>
        </ol>
        -->

       <h2>Citing RnBeads</h2>
       <p>
          Please cite RnBeads as follows:
          <ul>
            <li><a href="http://www.nature.com/nmeth/journal/vaop/ncurrent/full/nmeth.3115.html">Yassen Assenov*, Fabian M&uuml;ller*, Pavlo Lutsik*, J&ouml;rn Walter, Thomas Lengauer and Christoph Bock (2014) <i>Comprehensive Analysis of DNA Methylation Data with RnBeads</i>, Nature Methods, 11(11):1138-1140</a></li>
          </ul>
       </p>

       <h2>On the Shoulders of Giants</h2>

         
         <div class="row">
            <p>We would also like to emphasize that RnBeads makes extensive use of other R packages, and it may be appropriate to cite some of the underlying methods and contributing packages as well.</p>
         </div>

         <div class="row">
         <div class="col-md-3"></div>
         <div class="col-md-6">
            <img class="img-responsive img-rounded" src="images/dependencies.png" alt="">
         </div>
         <div class="col-md-3"></div>
         </div>
       
         <h4>Infinium data processing</h4>

         <ul>
            <li>Davis, S., Du, P., Bilke, S., Triche, T., Bootwalla, M. (2013) methylumi: Handle Illumina methylation data. R package version 2.6.1.</li>
         </ul>

         <h4>Filtering cross-reactive and SNP-overlapping probes</h4>

         <ul>
            <li>Chen, Y., Lemire, M., Choufani, S., Butcher, D.T.,  Grafodatskaya, D., Zanke, B.W., Gallinger, S., Hudson, T.J., Weksberg, R. (2013) Discovery of cross-reactive probes and polymorphic CpGs in the Illumina Infinium HumanMethylation450 microarray. Epigenetics, 8(2), 203-209</li>
            <li>Nordlund, J., B&auml;cklin, C. L. et al. (2013) Genome-wide signatures of differential DNA methylation in pediatric acute lymphoblastic leukemia. Genome Biology, 14:r105</li>
            <li>Pidsley, R., Zotenko, E., Peters, T.J., Lawrence, M.G., Risbridger, G.P., Molloy, P., Djik, S.V., Muhlhausler, B., Strizaker, C., Clark, S.J. (2016) Critical evaluation of the Illumina MethylationEPIC BeadChip microarray for whole-genome DNA methylation profiling. Genome Biology, 17(1), 208</li>
         </ul>

         <h4>Background correction</h4>

         <ul>
            <li>Triche, T.J., Jr., Weisenberger, D.J., Van Den Berg, D., Laird, P.W., Siegmund, K.D. (2013). Low-level processing of Illumina Infinium DNA Methylation BeadArrays. Nucleic Acids Res, 41(7), e90.</li>
         </ul>

         <h4>SWAN</h4>

         <ul>
            <li>Maksimovic, J., Gordon, L., Oshlack, A. (2012). SWAN: Subset quantile Within-Array Normalization for Illumina Infinium Infinium 450k BeadChips. Genome Biology, 13(6), R44.</li>
         </ul>

         <h4>Functional normalization (minfi.funnorm)</h4>

         <ul>
            <li>Fortin, J., Labbe, A., Lemire, M., Zanke, B.W., Hudson, T. J., Fertig, E.J., Greenwood, C.M.T., Hansen, K.D. (2014) Functional normalization of 450k methylation array data improves replication in large cancer studies. BioRxiv.</li>
         </ul>

         <h4>SWAN, functional normalization</h4>

         <ul>
            <li>Aryee, M.J., Jaffe, A.E., Corrada-Bravo, H., Ladd-Acosta, C., Feinberg, A.P., Hansen, K.D., Irizarry, R.A. (2014) Minfi: a flexible and comprehensive Bioconductor package for the analysis of Infinium DNA methylation microarrays. Bioinformatics 30(10), 1363-1369</li>
         </ul>

         <h4>BMIQ</h4>

         <ul>
            <li>Teschendorff, A.E., Marabita, F., Lechner, M., Bartlett, T., Tegner, J., Gomez-Cabrero, D., Beck, S. (2013) A beta-mixture quantile normalization method for correcting probe design bias in Illumina Infinium 450 k DNA methylation data.  Bioinformatics 29(2), 189-196.
         </ul>

         <h4>Data-driven normalization approaches</h4>

         <ul>
            <li>Pidsley, R., Wong, C., Volta, M., Lunnon, K., Mill, J., Schalkwyk, L. (2013)  A data-driven approach to preprocessing Illumina 450K methylation array data. BMC Genomics, 14(1), 293.</li>
            <li>Schalkwyk, L.C., Pidsley R., Wong C.C., Touleimat, W.C.F.N., Defrance M., Teschendorff A., Maksimovic J. (2013). wateRmelon: Illumina 450 methylation array normalization and metrics. R package version 1.4.0.</li>
         </ul>

         <h4>Genome-wide coverage plots in quality control for bisulfite sequencing data</h4>

         <ul>
            <li>Yin, T., Cook, D., Lawrence, M. (2012). ggbio: an R package for extending the grammar of graphics for genomic data. Genome Biology, 13(8), pp. R77.</li>
         </ul>

         <h4>Differential methylation using the &quot;limma&quot; method</h4>

         <ul>
            <li>Smyth, G.K. (2005). Limma: linear models for microarray data. In Gentleman R, Carey, V., Dudoit., S., Irizarry R., Huber W. (eds.), Bioinformatics and Computational Biology Solutions Using R and Bioconductor, pp. 397-420. Springer, New York.</li>
         </ul>

         <h4>Reference-based cell type heterogeneity estimation</h4>

         <ul>
            <li>Houseman, E.A., Accomando, W.P., Koestler, D.C., Christensen, B.C., Marsit, C.J., Nelson, H.H., Wiencke, J.K., Kelsey, K.T. (2012). DNA methylation arrays as surrogate measures of cell mixture distribution. BMC Bioinformatics 13:86.</li>
         </ul>

         <h4>Reference-free correction for cell type heterogeneity</h4>

         <ul>
            <li>Houseman, E.A., Molitor, J., Marsit, C.J. (2014). Reference-Free Cell Mixture Adjustments in Analysis of DNA Methylation Data. Bioinformatics, btu029.</li>
         </ul>

    </div>
    <!-- /.container -->

    <!-- Footer -->
    <hr>
    <footer class="py-5 bg-inverse">
        <div class="container">
            <p class="m-0 text-center text-white">Copyright &copy; <a href="mailto:rnbeads@mpi-inf.mpg.de">RnBeads Development Team</a> 2019 | <a href="https://imprint.mpi-klsb.mpg.de/inf/www.mpi-inf.mpg.de">Impressum/Imprint</a> | <a href="https://data-protection.mpi-klsb.mpg.de/inf/www.mpi-inf.mpg.de">Datenschutz/Data protection</a></p>
        </div>
    </footer>
    <!-- jQuery -->
    <script src="js/jquery.js"></script>

    <!-- Bootstrap Core JavaScript -->
    <script src="js/bootstrap.min.js"></script>


</body>

</html>
